src/WordPress/JsonMapper/Evaluators/PropertyMapper.php

```diff
@@ -0,0 +1,289 @@
+<?php
+
+namespace WordPress\JsonMapper\Evaluators;
+
+use ReflectionClass;
+use ReflectionMethod;
+use WordPress\Blueprints\Model\DataClass\ModelInfo;
+use WordPress\JsonMapper\JsonMapper;
+use WordPress\JsonMapper\JsonMapperException;
+use WordPress\JsonMapper\ObjectWrapper;
+use WordPress\JsonMapper\Property\Property;
+use WordPress\JsonMapper\Property\PropertyMap;
+use WordPress\JsonMapper\Property\PropertyType;
+use function in_array;
+
+class PropertyMapper implements JsonEvaluatorInterface {
+	/**
+	 * @var string[]
+	 */
+	private $scalar_types = array( 'string', 'bool', 'boolean', 'int', 'integer', 'double', 'float' );
+
+	/**
+	 * @var callable[]
+	 */
+	private $factories = array();
+
+	/**
+	 * @var JsonMapper
+	 */
+	private $mapper;
+
+	public function __construct(
+		JsonMapper $json_mapper
+	) {
+		$this->mapper = $json_mapper;
+		$this->add_factories_for_native_php_classes();
+		$this->add_factory_for_resources();
```

Comment:
This registers Blueprint-specific handlers in a `PropertyMapper` that looks generic. How easy would it be to move the registration of specific handlers to `BlueprintMapper`? And then, perhaps, the Mapper library would eventually become generic like `WordPress/JsonMapper`.

Reply:
Easy enough. So easy in fact that it can be done when it is needed. For now, with no other application of the Json Mapper in sight - the code rolls quite nicely as it is, no additional variables need to be passed when instantiating a specific factory, as all are instantiated as private methods in this class.

In fact, with what the mapper is now, I wonder that purpose do those two classes have: `BlueprintMapper` and `JsonMapper`. With a generic mapper - sure, they should be separate. But for now?

Reply:
With this structure, maintaining these handlers will require diving into internals of this JSON mapper. It's counter-intuitive to look for implementation-specific handlers in a class that seems generic. If it's as easy, let's just do it. Discussing, creating tracking issues, navigating the codebase will likely take more time than just making this change.

Reply:
Done. [See this commit.](https://github.com/WordPress/blueprints/pull/69/commits/da346af421e0c42c25d190799e03f60bee3456cb)